src/main/java/org/example/models/PlayerStatus.java

```diff
@@ -31,8 +35,9 @@ public WeaponModel getWeaponInHand() {
 
     public void setWeaponInHand(WeaponModel weapon, Weapons weaponType) {
         weaponInHand = new WeaponModel(FIST);
-        if (!canBuyWeapon(weapon, score)) {
+        if (service.canBuyWeapon(weapon, score)) {
             weaponInHand.setName(weaponType);
```

Comment:
To simplify, I'd modify the 'name' property to be of type 'Weapons' enum, instead of String and, maybe, rename it to 'weapon' or 'weaponName'.